bin/faucet/src/main.rs

```diff
@@ -72,6 +73,26 @@ async fn main() -> Result<(), FaucetError> {
 
             info!(target: COMPONENT, %config, "Initializing server");
 
+            let client_state_clone = Arc::clone(&faucet_state.client);
+
+            info!("Initializing chain tip updater");
+            thread::spawn(move || {
+                let sys = actix_web::rt::System::new();
+                sys.block_on(async move {
+                    loop {
+                        let state_clone_inner = Arc::clone(&client_state_clone);
+                        actix_web::rt::spawn(async move {
```

Comment:
I'm not very familiar with actix, but do you need to spawn here? Isn't it already in a future so you could just call it without the additional spawn?

Reply:
Oh, no. You are right. This spawn is not needed. I'm removing it. Thanks!